src/Notes/WC_Admin_Notes_Facebook_Extension.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * WooCommerce Admin Facebook Extension Note Provider.
+ *
+ * Adds a note to the merchant's inbox showing the benefits of the Facebook extension.
+ *
+ * @package WooCommerce Admin
+ */
+
+namespace Automattic\WooCommerce\Admin\Notes;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Admin_Notes_Facebook_Extension
+ */
+class WC_Admin_Notes_Facebook_Extension {
+
+	/**
+	 * Name of the note for use in the database.
+	 */
+	const NOTE_NAME = 'wc-admin-facebook-extension';
+
+	/**
+	 * Attach hooks.
+	 */
+	public function __construct() {
+		add_action( 'woocommerce_admin_note_action_install-now', array( $this, 'install_facebook_extension' ) );
+	}
+
+	/**
+	 * Possibly add Facebook extension note.
+	 */
+	public static function possibly_add_facebook_note() {
+		$wc_admin_installed = get_option( 'wc_admin_install_timestamp', false );
+		if ( false === $wc_admin_installed ) {
```

Comment:
Optional: I notice we use this a few times across different notes; might be nice to reuse some of this logic (terrible name, but general idea):

```php
	public static function is_wc_admin_installed_for( $seconds ) {
		$wc_admin_installed = get_option( 'wc_admin_install_timestamp', false );
		if ( false === $wc_admin_installed ) {
			$wc_admin_installed = time();
			update_option( 'wc_admin_install_timestamp', $wc_admin_installed );
		}

		$current_time        = time();

		return $current_time - $wc_admin_installed < $seconds );
	}
```

Reply:
Good idea - how about introducing a `Trait` class for notes?

Reply:
Done in a810cc9.